engine/src/test/java/io/camunda/zeebe/engine/processing/processinstance/CreateProcessInstanceAnywhereTest.java

```diff
@@ -378,11 +378,7 @@ public void shouldCreateVariablesInProcessScope() {
     // Then
     Assertions.assertThat(
             RecordingExporter.records()
-                .limit(
-                    // extract the condition into a method like limitToProcessInstance()
-                    r ->
-                        r.getKey() == processInstanceKey
-                            && r.getIntent() == ProcessInstanceIntent.ELEMENT_COMPLETED)
+                .limitToProcessInstance(processInstanceKey)
```

Comment:
:question: I don't understand how this works, to be honest. The `limitToProcessInstance` gives me all records where the key is the process instance (and before it is being closed). I would assume this predicate is positive for all process instance records. But somehow there seem to be variable records among them, or else this test doesn't work.
Are the variable records which have as key the process instance??? Because the key on a variable records points to a scope and the key on a process instance record is the key of the process instance?

Overall, I find this difficult to comprehend. This test is about the variables that are being written. Sp is there no way to rewrite this as:

`
RecordingExporter.variableRecords().withEventScopeKey(processInstanceKey).withIntent(VariableIntent.CREATED).findFirst()
`

Reply:
>  I don't understand how this works, to be honest. The limitToProcessInstance gives me all records where the key is the process instance

No, your assumption is not correct. The method uses the `limit()` function of the stream. It returns all records (of all types) until the `completed` event of the process instance.

> is there no way to rewrite this as:

Sure, we could write the condition in this way. But this doesn't verify that any other variable not written, for example, for another scope.